hpe3parclient/client.py

```diff
@@ -4872,3 +4792,252 @@ def getScheduleStatus(self, schedule_name):
+            is not a base volume.
+        :raises: :class:`~hpe3parclient.exceptions.HTTPForbidden`
+            - INV_OPERATION_VV_SYS_VOLUME - The volume is a system volume. This
+            operation is not allowed on a system volume.
+        :raises: :class:`~hpe3parclient.exceptions.HTTPForbidden`
+            - INV_OPERATION_VV_CLEANUP_IN_PROGRESS - Internal volume cleanup is
+            in progress.
+        :raises: :class:`~hpe3parclient.exceptions.HTTPForbidden`
+            - INV_OPERATION_VV_INTERNAL_VOLUME - Cannot modify an internal
+            volume
+        :raises: :class:`~hpe3parclient.exceptions.HTTPConflict`
+            - INV_OPERATION_VV_VOLUME_CONV_IN_PROGRESS - Invalid operation: VV
+            conversion is in progress.
+        :raises: :class:`~hpe3parclient.exceptions.HTTPForbidden`
+            - INV_OPERATION_VV_NOT_IN_NORMAL_STATE - Volume state is not normal
+        :raises: :class:`~hpe3parclient.exceptions.HTTPForbidden`
+            - INV_OPERATION_VV_PEER_VOLUME - Cannot modify a peer volume.
+        :raises: :class:`~hpe3parclient.exceptions.HTTPConflict`
+            - INV_OPERATION_VV_TASK_CANCEL_IN_PROGRESS - Invalid operation:
+            A task involving the volume is being canceled..
+        :raises: :class:`~hpe3parclient.exceptions.HTTPConflict`
+            - INV_OPERATION_VV_PROMOTE_IN_PROGRESS - Invalid operation: Volume
+            promotion is in progress.
+        :raises: :class:`~hpe3parclient.exceptions.HTTPConflict`
+            - INV_OPERATION_VV_TUNE_IN_PROGRESS - Invalid operation: Volume
+            tuning is in progress.
+        :raises: :class:`~hpe3parclient.exceptions.HTTPBadRequest`
+            - NO_SPACE - Not Enough space is available
+        :raises: :class:`~hpe3parclient.exceptions.HTTPForbidden`
+            - NODE_DOWN - The node is down.
+        :raises: :class:`~hpe3parclient.exceptions.HTTPNotFound`
+            - NON_EXISTENT_CPG - The CPG does not exists.
+        :raises: :class:`~hpe3parclient.exceptions.HTTPNotFound`
+            - NON_EXISTENT_VOL - volume doesn't exist
+        :raises: :class:`~hpe3parclient.exceptions.HTTPForbidden`
+            - VV_IN_INCONSISTENT_STATE - The volume has an internal consistency
+            error.
+        :raises: :class:`~hpe3parclient.exceptions.HTTPForbidden`
+            - VV_IS_BEING_REMOVED - The volume is being removed.
+        :raises: :class:`~hpe3parclient.exceptions.HTTPForbidden`
+            - VV_NEEDS_TO_BE_CHECKED - The volume needs to be checked.
+        :raises: :class:`~hpe3parclient.exceptions.HTTPForbidden`
+            - VV_NOT_STARTED - Volume is not started.
+        :raises: :class:`~hpe3parclient.exceptions.HTTPForbidden`
+            - INV_INPUT_VV_IS_FPVV - A fully provisioned volume cannot be
+            compressed.
+
+        """
+        info = {'action': self.TUNE_VOLUME, 'tuneOperation': tune_operation}
+        if optional is not None and not self.compression_supported:
```

Comment:
@ansarars 
is it better if we do not delete the compression entry and throw proper error while tuning if compression is passed as True and WSAPI doe not support compression,
We can make user aware what option can be passed with what WSAPI versions.

Reply:
> @ansarars
> is it better if we do not delete the compression entry and throw proper error while tuning if compression is passed as True and WSAPI doe not support compression,
> We can make user aware what option can be passed with what WSAPI versions.
@nilangekarss : If we convert volume type from thin to full on 3.2.2 array, then by default
compression=false comes from ansible, which result failure from array since compression key
is not supported on 3.2.2. Hence we cant remove code for removing compression key from optional for 3.2.2 array otherwise thin to full or thin to dedupe will not work due to compression key error.